### PR TITLE
Scoring System

### PR DESCRIPTION
**Pull request by ChaseBENNC9**
## Describe your changes

This is going to add in a UI for the score and a timer for the level, the visuals are temporary but should be easily transferable to the updated UI. The score of each team will increase when they score points, the timer will start from a specified value. when the timer runs out or a target score is reached by either team the match will end. 
If I have enough time i will also include the "End Game System" which covers showing the winners, some statistics to the screen.

## Issue number and link

#24 

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: Assets/Scripts/LevelManager.cs [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+        enemyScoreTitle.text = GetTeamName(enemyTeam);
+
+        timerTitle.text = "Starting in ";
+        timerText.text = delayTime.ToString();
+        currentTime = Time.time;
+    }
+
+    void Update()
+    {
+        {
```
Comment (PBWiseman) [inline Assets/Scripts/LevelManager.cs:95]: Extra bracket here.

File: Assets/Scripts/LevelManager.cs [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+    }
+
+    /// <summary>
+    /// This method will return the name of the team based on the enum value
+    /// </summary>
+    /// <param name="team">The Team to be returned as a string</param>
+    /// <returns>The specified team as a string</returns>
+    private string GetTeamName(Teams team)
+    {
+        if (team == Teams.Penguins)
```
Comment (PBWiseman) [inline Assets/Scripts/LevelManager.cs:156]: Would this be better as a switch statement?

**Review by PBWiseman — CHANGES_REQUESTED** [review]
# Code Review

## Functional Requirements
_Is the application fit for purpose? Does it meet the user’s requirements and properly address the issue?_  

 - Feature looks good! Everything works well.


## Legibility and Style
_Uses agreed coding standards/conventions? Readable, consistent and well-structured with useful comments?_  

 - My only concern is that it might be better practice to use a game state enum instead of the round over and round started bool? Would there be issues with that? It could also incorporate a paused state and it would mean that you could just check if the game is in the active enum rather than checking if each different bool is set correctly? Also the variable could go in GameSettings so be better scoped for something that will be likely used by a lot of different files.

## Implementation
_Any notable performance/speed issues? Compatibility and Responsiveness on other platforms ?_  

 - None


# Final Comments

 - Would like to hear your thoughts about the enum idea. Other than that and a couple of minor formatting issues I am happy for this to be merged.

***

**Comment by PBWiseman** [comment]
Another issue I just found is that sometime the score still updates after 15 is reached if the snowball is in the air at the time.
image
There were many more than 2 in the air but only the first couple counted and then it stopped counting.

**Comment by ChaseBENNC9** [comment]
That should cover both the issues.

The enum has been implemented and I believe it should be functioning as expected. 

The score should no longer go above 15 or essentially whatever the target score is set to.

I haven't yet had a look at the enemy movement issue but will get to that.

**Review by PBWiseman — APPROVED** [review]
Concerns addressed. Happy for this to be merged

File: Assets/Scripts/LevelManager.cs [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+                if (delayTime > 0)
+                {
+                    currentTime = Time.time;
+                    delayTime -= 1;
+                    timerTitle.text = "Starting in ";
+                    timerText.text = delayTime.ToString();
+                }
+                else
+                {
+                    // if (!roundStarted)
```
Comment (erokasyuart) [inline Assets/Scripts/LevelManager.cs:107]: Commented code needed?

File: Assets/Scripts/LevelManager.cs [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+            }
+        }
+    }
+
+    /// <summary>
+    /// This method will return the name of the team based on the enum value
+    /// </summary>
+    /// <param name="team">The Team to be returned as a string</param>
+    /// <returns>The specified team as a string</returns>
+    private string GetTeamName(Teams team)
```
Comment (erokasyuart) [inline Assets/Scripts/LevelManager.cs:153]: I have to agree, this would be a good place for a switch!

File: Assets/Scripts/LevelManager.cs [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+        {
+            return "Unknown";
+        }
+    }
+
+    /// <summary>
+    /// This method will display the winner of the match
+    /// </summary>
+    /// <param name="winner">The team that won the match</param>
+    private void DisplayWinner(string winner)
```
Comment (erokasyuart) [inline Assets/Scripts/LevelManager.cs:181]: Same with this! I can see this as a switch too.

**Review by erokasyuart — CHANGES_REQUESTED** [review]
Looks pretty good otherwise!

**Review by erokasyuart — APPROVED** [review]
Looks good.

**Pull request merged**